### PR TITLE
Update to Gradle 7.6

### PR DESCRIPTION
**Pull request by altrisi**
This PR updates Gradle and the wrapper to 7.6. This is the result of running the command `gradlew wrapper --gradle-version=7.6`.

I wanted to update the wrapper too given 7.6 fixed a [5 year old issue](https://togithub.com/gradle/gradle/issues/3605) where the wrapper would also keep the zip of every used Gradle version after extracting it, with each zip being ~100MB (over that in recent versions). It should now be deleted after extraction ([gradle/gradle#19495](https://github.com/gradle/gradle/pull/19495)).

Doing so not only uses less space in modder's drives, but also can (slightly) speed up github actions cache downloads given that zip can be a significant part of the cache (around 25% in carpet for example unless I was missing something).

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.5.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.6-bin.zip
+networkTimeout=10000
```
Comment (modmuss50) [inline gradle/wrapper/gradle-wrapper.properties:4]: Is this something you added, when I ran `./gradlew wrapper --gradle-version=7.6` in another project this was not added.

**Comment by modmuss50** [comment]
Something isnt right with this PR, when I ran the upgrade on my machine I got a different result: https://github.com/FabricMC/fabric-example-mod/pull/223 This PR updates the wrapper jar, while mine does not?

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (altrisi) [reply gradle/wrapper/gradle-wrapper.properties:4]: It was added for me, IIRC in all projects I ran it in, not just here. It was added in 7.6 according to the gradle repo too.

**Comment by altrisi** [comment]
Strange, quite sure I changed nothing, feel free to just push that then, tbh all I really want is for that bug to not affect me or anyone when starting a new mod.

PS: Maybe the diff is different because it was made from a different origin version? Idk

**Comment by modmuss50** [comment]
Ah, figured it out, I first ran it from 7.5, re-running the same command from 7.6 gives the same results that you got. That is super weird...

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**